SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -200,31 +203,73 @@ private Task ProcessPendingErrorsAsync()
 
         private Task SendCrashReportsOrAwaitUserConfirmationAsync()
         {
-            return HandleUserConfirmationAsync();
+            bool alwaysSend = ApplicationSettings.GetValue(PrefKeyAlwaysSend, false);
+
+            return Task.Run(async () =>
+            {
+                if (_unprocessedManagedErrorLogs.Count() > 0)
```

Comment:
Has thread safety been considered? (I'm not suggesting that it's wrong, nor that there should be a lock around this.)